### DISCUSSION TITLE
How did you manage get the tts Python package  into an exe?😅

### DISCUSSION THREAD

**Discussion by DrewThomasson** [lukaszliniewicz/Pandrator#49] [Q&A] [answered]
I keep running into issues lol

**Comment by lukaszliniewicz** [comment, marked as answer]
It's just the installer/launcher, not the whole thing - it creates conda environments and installs different packages based on user choice. For an all-in-one package I went with compressing the launcher and the folder with conda envs/dependencies into **[archives](https://github.com/lukaszliniewicz/Pandrator#self-contained-packages)** with various "preinstalled" components... It's not very elegant, but easy to use.

**Comment by DrewThomasson** [comment]
Hmm

Thx for the info tho 

🤔

**Comment by lukaszliniewicz** [comment]
On a completely different topic - did you manage to get StyleTTS2 with the replaced phonemiser to work well on Windows? It was unusable for me, mispronounced virtually everything...

**Reply by DrewThomasson** [reply]
Well I dockerize everything I release at this point so that makes it universal lol


But also I just use the styletts2 pip installer

https://github.com/sidharthrajaram/StyleTTS2

- I don't know if it has your issue on windows though but it's worked on everything else for me

**Reply by DrewThomasson** [reply]
Idk if that helps but yeah lol

I also had to fix some dependency conflicts with it and the coqui tts pip but was relatively simple lol

**Reply by lukaszliniewicz** [reply]
I just keep everything in its own conda env. It's bloated, sure, but I'd rather do that than deal with dependency conflicts ; ) The pip package was the one I tried. I thought the problems might be related to the phonemiser (it uses gruut instead of espeak-ng, I think), but if it works ok for you in a Docker container, then it was not the issue. I will probably look into it at some point, but for now, I think XTTS is good enough, especially fine-tuned.

**Reply by DrewThomasson** [reply]
Honestly agreed

Xtts is so good that even given the choice on CPU only people still go for Xtts over styletts2 lol

**Reply by lukaszliniewicz** [reply]
Yes, it's very good, and can be made even better with fine-tuning on 20m of good quality audio, which is simple and not THAT resource intensive.

**Reply by DrewThomasson** [reply]
Honestly yup

Pretty insane that all you need is a like a cpu only computer with 16 gb ram lol
